lib/core/src/Cardano/Wallet.hs

```diff
@@ -985,6 +986,43 @@ signPayment ctx wid argGenChange pwd (CoinSelection ins outs chgs) = db & \DBLay
     db = ctx ^. dbLayer @s @k
     tl = ctx ^. transactionLayer @t @k
 
+-- | Augments the given set of outputs with additional outputs for each of the
+--   given change outputs.
+--
+-- This function associates each change value with a target address belonging
+-- to the given wallet, updates the wallet state to keep track of pending
+-- change addresses, and writes a database checkpoint to record the updated
+-- state.
+--
+-- See 'assignChangeAddresses'.
+--
+assignChangeAddressesAndCheckpoint
```

Comment:
**[change]**

Rename this to (for example) `selectCoinsForExternal` and move the logic from the API handler to here.

**Why?** 

Having to define errors type for the Api handlers is a good sign that some logic is in the wrong place. We should keep the API handlers simple and move as much as the wallet logic down to this module. The API layer should only compose functions from the wallet engine. Plus, the name `assignChangeAddressesAndCheckpoint` already suggest that something seems wrong (everytime one is tempted to put `And` in the name of a function is a good sign that this function is probably not operating at the right level of abstraction).

Reply:
> Rename this to (for example) selectCoinsForExternal and move the logic from the API handler to here.

Fair enough! I've made the following change to implement your suggestion: https://github.com/input-output-hk/cardano-wallet/pull/1165/commits/62b45c411171f5e91ea044f50e48144596faf78b.

Of course, let me know if this is not what you had in mind.

As a result the above change, the tuple `(NonEmpty (TxIn, TxOut), NonEmpty TxOut)` appears in the signatures of both the following functions:

* `selectCoinsForExternal`
* `mkApiCoinSelection`

To simplify this, I've made a further commit which replaces this tuple with a dedicated type, to capture the concept of _**a coin selection that is ready to be signed**_: https://github.com/input-output-hk/cardano-wallet/pull/1165/commits/437a0a83f51972f73d0296d0dda1a8f68c177705.

Let me know if this is suitable. If not, I'd be happy to revise it further. Thanks in advance!